pluma/pluma.c

```diff
@@ -515,7 +525,7 @@ main (int argc, char *argv[])
 	bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
 	textdomain (GETTEXT_PACKAGE);
 
-	gdk_set_allowed_backends ("x11");
+	gdk_set_allowed_backends ("wayland,x11");
```

Comment:
You should use HAVE_X11 and HAVE_WAYLAND (Set allowed backends, https://discourse.ubuntu.com/t/porting-mate-apps-to-wayland/12670) for setting allowed backends.
```
#if defined(HAVE_X11) && defined(HAVE_WAYLAND)
    gdk_set_allowed_backends ("wayland,x11");
#elif defined(HAVE_WAYLAND)
    gdk_set_allowed_backends ("wayland");
#else
    gdk_set_allowed_backends ("x11");
#endif
```

Indeed, first you should add the snippet below in the configure.ac file. https://raw.githubusercontent.com/mate-desktop/mate-panel/8146d0c4f841d45a812a0408028b7af167428c6b/configure.ac

```
# $enable_x11 and $enable_wayland will be set to "yes", "no" or "auto"
AC_ARG_ENABLE(x11,
              [AC_HELP_STRING([--enable-x11],
                              [Explicitly enable or disable X11 support
                              (default is to enable only if X development libraries are detected)])],
              [enable_x11=$enableval],
              [enable_x11=auto])

AC_ARG_ENABLE(wayland,
              [AC_HELP_STRING([--enable-wayland],
                              [Explicitly enable or disable Wayland support
                              (default is to enable only if Wayland client development library is detected)])],
              [enable_wayland=$enableval],
              [enable_wayland=auto])

# Check if we have gtk-layer-shell installed, and thus should build with Wayland support
have_wayland=no
if test "x$enable_wayland" != "xno"; then
  PKG_CHECK_MODULES(GTK_LAYER_SHELL, gtk-layer-shell-0, have_wayland=yes, [
        if test "x$enable_wayland" = "xyes"; then
          AC_MSG_ERROR([Wayland enabled but GTK Layer Shell library not found])
        fi
    ])
fi

AM_CONDITIONAL(ENABLE_WAYLAND, [test "x$have_wayland" = "xyes"])

if test "x$have_wayland" = "xyes"; then
  AC_DEFINE(HAVE_WAYLAND, 1, [Have the Wayland development library])
fi

AC_SUBST(GTK_LAYER_SHELL_CFLAGS)
AC_SUBST(GTK_LAYER_SHELL_LIBS)

# Check if we have the X development libraries
have_x11=no
if test "x$enable_x11" != "xno"; then
  PKG_CHECK_MODULES(X, x11 xau, have_x11=yes, [
      if test "x$enable_x11" = "xyes"; then
        AC_MSG_ERROR([X development libraries not found])
      fi
    ])
fi

AM_CONDITIONAL(ENABLE_X11, [test "x$have_x11" = "xyes"])

if test "x$have_x11" = "xyes"; then
  AC_DEFINE(HAVE_X11, 1, [Have the X11 development library])
fi

if test "x$have_x11" != "xyes" -a "x$have_wayland" != "xyes"; then
  if test "x$enable_wayland" = "xno" -a "x$enable_x11" = "xno"; then
    AC_MSG_ERROR([At least one backend must be enabled])
  else
    AC_MSG_ERROR([No usable backend found, install X11 or Wayland development libraries])
  fi
fi

AC_SUBST(X_LIBS)
```
Then you'd check the config.h file after performing ./autogen.sh or configure with --enable-wayland and/or  --enable-x11 options.